### ISSUE TITLE
alpine build failure:  undefined reference to makedev

### ISSUE THREAD

**Issue by truatpasteurdotfr** [apptainer/singularity#223]
alpine 3.4 build failing with:

```
libtool: link: gcc -Wall -fpie -fPIC -g -O2 -Wl,-rpath -Wl,/usr/local/lib -o sexec-suid sexec_suid-sexec.o util/sexec_suid-util.o util/sexec_suid-file.o  lib/.libs/libsingularity.a
lib/.libs/libsingularity.a(dev.o): In function `singularity_mount_dev':
/singularity/src/lib/mount/dev/dev.c:77: undefined reference to `makedev'
/singularity/src/lib/mount/dev/dev.c:70: undefined reference to `makedev'
/singularity/src/lib/mount/dev/dev.c:63: undefined reference to `makedev'
/singularity/src/lib/mount/dev/dev.c:56: undefined reference to `makedev'
collect2: error: ld returned 1 exit status
make[3]: *** [sexec-suid] Error 1
Makefile:544: recipe for target 'sexec-suid' failed
make[3]: Leaving directory '/singularity/src'
Makefile:645: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/singularity/src'
Makefile:407: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/singularity/src'
Makefile:397: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by truatpasteurdotfr** [comment]
git version as of today fails but singularity-2.1.2 build fine.

**Comment by truatpasteurdotfr** [comment]
single line fix

```
diff --git a/src/lib/mount/dev/dev.c b/src/lib/mount/dev/dev.c
index 7052020..146de4f 100644
--- a/src/lib/mount/dev/dev.c
+++ b/src/lib/mount/dev/dev.c
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/types.h>
 #include <sys/mount.h>
 #include <unistd.h>
 #include <stdlib.h>
```

**Comment by gmkurtzer** [comment]
Fixed in [master b7febad]. Thanks!
